### PR TITLE
Blog App: Validations and model specs

### PR DESCRIPTION
**Pull request by Mike111222**
- Add the following validations:

- For the User model:

Name must not be blank.
PostsCounter must be an integer greater than or equal to zero.

- For the Post model:

Title must not be blank.
Title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.

- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+      @post.likes_counter = 1
+      expect(@post).to be_valid
+    end
+
+    it 'should return less than 5 comments ' do
+      value = @post.recent_comments.length
+      expect(value).to be < 5
+    end
+  end
```
Comment (nahnahdev) [inline spec/post_spec.rb]: - [x] Great job so far @Mike111222 There are a few changes I will suggest you implement and you'll be on your way to the next Milestone :wink: 
- [x] Great job adding most of the tests, however, I noticed that per the project requirements, you were asked to `Add unit specs for all of your models' methods and validations.` But currently, there are some methods in your `Posts Model` that haven't been tested. So kindly include tests for those methods as well :wink: 
- [x] Methods Yet to Be Tested
```ruby
  def update_author_posts_counter
    author.update_posts_counter
  end
```

File: app/models/user.rb [hunk 2]
```diff
@@ -3,11 +3,18 @@ class User < ApplicationRecord
...
   end
 
   def update_posts_counter
     update(posts_counter: posts.count)
   end
+
+  def update_comments_counter
+    update(comments_counter: comments.count)
+  end
 end
```
Comment (nahnahdev) [inline app/models/user.rb]: - [x] I noticed that no tests have been added for your `User Model`. So kindly ensure that you `Add unit specs for all the methods and validations` in your `User Model` :100:

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @Mike111222  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- All required user validations added :heavy_check_mark: 
- All required posts validations added :heavy_check_mark: 
- Has unit tests for  `Post`, `Comment`, and `Like` model :heavy_check_mark: 
- All tests passing :heavy_check_mark: 
- Model specs has **Validation** tests :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/post_spec.rb (hunk 1 above)
Comment (Mike111222) [reply spec/post_spec.rb]: I have made the changes you suggested. Thanks for the kind feedback.

File: app/models/user.rb (hunk 2 above)
Comment (Mike111222) [reply app/models/user.rb]: I have made the changes you suggested. Thanks for the kind feedback.

**Review by frank1738 — APPROVED** [review]
Hi @Mike111222 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Good job including validation tests
✅ Profesional documentation with meaningful commit messages 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**